modules/programs/firefox.nix

```diff
@@ -40,6 +60,77 @@ in
         '';
       };
 
+      profiles = mkOption {
+        type = types.attrsOf (types.submodule ({config, name, ...}: {
+          options = {
+            name = mkOption {
+              type = types.str;
+              default = name;
+              description = "Profile name.";
+            };
+
+            id = mkOption {
+              type = types.int;
+              default = 0;
+              description = "Profile ID.";
+            };
+
+            preferences = mkOption {
+              type = types.attrs;
+              default = {};
+              example = {
+                "browser.startup.homepage" = "https://nixos.org";
+                "browser.search.region" = "GB";
+                "browser.search.isUS" = false;
+                "distribution.searchplugins.defaultLocale" = "en-GB";
+                "general.useragent.locale" = "en-GB";
+                "browser.bookmarks.showMobileBookmarks" = true;
+              };
+              description = "Attribute set of firefox preferences.";
+            };
+
+            extraPreferences = mkOption {
+              type = types.lines;
+              default = "";
+              description = "Extra preferences";
+            };
+
+            customCSS = mkOption {
+              type = types.lines;
+              default = "";
+              description = "Custom firefox CSS.";
+              example = ''
+                /* Hide tab bar in FF Quantum */
+                @-moz-document url("chrome://browser/content/browser.xul") {
+                  #TabsToolbar {
+                    visibility: collapse !important;
+                    margin-bottom: 21px !important;
+                  }
+
+                  #sidebar-box[sidebarcommand="treestyletab_piro_sakura_ne_jp-sidebar-action"] #sidebar-header {
+                    visibility: collapse !important;
+                  }
+                }
+              '';
+            };
+
+            path = mkOption {
+              type = types.str;
+              default = name;
+              description = "Profile path.";
+            };
+
+            isDefault = mkOption {
+              type = types.bool;
+              default = config.id == 0;
```

Comment:
Maybe add a `defaultText` attribute, I believe this default value will cause the manual to depend on the configuration.

Something like
```
defaultText = "true if profile ID is 0";
```
perhaps?

Reply:
Done